app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/MapScreen.kt

```diff
@@ -1,13 +1,41 @@
 package ch.epfl.cs311.wanderwave.ui.screens
 
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.platform.testTag
 import androidx.compose.ui.tooling.preview.Preview
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import ch.epfl.cs311.wanderwave.viewmodel.MapViewModel
 import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.Marker
+import com.google.maps.android.compose.MarkerState
 
 @Composable
 @Preview
 fun MapScreen() {
-  GoogleMap(modifier = Modifier.testTag("mapScreen"))
+  val viewModel: MapViewModel = hiltViewModel()
+  val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+
+  //TODO: Make sure the beacons are filtered to be within the radius
+  GoogleMap(modifier = Modifier.testTag("mapScreen")) { DisplayBeacons(beacons = uiState.beacons) }
```

Comment:
Does this actually work like this? Shouldn't we wait until `uiState.loading` is false until we display the beacons?

Reply:
You're right, corrected it!